### PR TITLE
PLAT-1032 Auto-complete inputs: Ignore diacritics

### PR DESCRIPTION
**Pull request by alexander-schmidt**
Peek 2022-10-13 20-21

### CODE REVIEW

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java [hunk 1]
```diff
@@ -18,10 +19,11 @@
  * performed, by appending numeric suffixes to the respective display strings,
  * e.g. "Foo (1)", "Foo (2)", etc.
  * <p>
- * Comparison between display strings is case-insensitive, meaning that display
- * strings only varying in letter cases are also deduplicated, e.g. "foo" and
- * "Foo". However, letter cases are retained when suffixes are added, resulting
- * in "foo (1)" and "Foo (2)".
+ * Comparison between display strings is case-insensitive and
+ * diacritic-agnostic, meaning that display strings only varying in letter cases
+ * and diacritics are also deduplicated, e.g. "foo" and "Foo", or "foo" and
```
Comment (oliver-richers) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java:24]:
```suggestion
 * and/or diacritics are also deduplicated, e.g. "foo" and "Foo", or "foo" and
```

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java [hunk 2]
```diff
@@ -32,19 +34,23 @@
 
 	private final Function<T, IDisplayString> displayFunction;
 	private final Comparator<T> comparator;
```
Comment (oliver-richers) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java:36]: maybe rename `comparator` to `valueComparator` and put it next to `keyComparator`?

**Review by oliver-richers — APPROVED** [review]

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java:24]: ok

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java (hunk 2 above)
Comment (alexander-schmidt) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDisplayStringDeduplicator.java:36]: ok

**Pull request merged**